### PR TITLE
fix: expand and reset

### PR DESCRIPTION
**Pull request by davec504**
* chore: prerelease

* Chore: prerelease

Release-As: 0.0.2-rc1

* fix: test changes

* Chore: prerelease

Release-As: 0.0.2-rc1

* fix: various changes

* Chore: prerelease

Release-As: 0.0.2-rc3

* fix: add extra drag

* fix: filter colour changes

---------

Signed-off-by: BMPMS-Telicent <135822574+BMPMS-Telicent@users.noreply.github.com>

### CODE REVIEW

File: src/components/OntologyHierarchy/OntologyHierarchy.tsx [hunk 1]
```diff
@@ -13,52 +13,58 @@ export interface OntologyHierarchyProps {
...
+      if(
+          JSON.stringify(filterIds) === JSON.stringify(currentFilterIds.current) &&
+          baseKey === currentBaseKey.current &&
+          descendantCount === currentDescendantCount.current &&
+          expandElement === currentExpandElement.current &&
+          expandAll === currentExpandAll.current
+      ){
+          // no change so don't re-render
+          return
+      } else {
```
Comment (ACoolmanTelicent) [inline src/components/OntologyHierarchy/OntologyHierarchy.tsx:61]: obs: `else` is not needed

**Review by ACoolmanTelicent — DISMISSED** [review]

**Comment by ACoolmanTelicent** [comment]
As I understand it, this has been tested pretty well by Bryony and Vanessa

**Comment by ACoolmanTelicent** [comment]
I _think_ the release workflow will work when this is merged.

File: src/components/OntologyHierarchy/OntologyHierarchy.tsx [hunk 2]
```diff
@@ -186,36 +244,32 @@ const OntologyHierarchy: React.FC<OntologyHierarchyProps> = ({
...
+          .text((d) => d.data.expandLabel || "")
 
       treeGroup.select(".expandLabelIcon")
+          .attr("pointer-events", "none")
         .attr("text-anchor", "middle")
         .attr("class", (d) => `expandLabelIcon ${!d.data.ontology ? "" : d.data.ontology.faIcon}`)
-        .attr("x", (d) => (d.data.startLeft || 0)  + 7.5 + (d.depth === 1 ? rowHeight - 3 : rowHeight - 6))
-        .attr("y", (d) => (d.data.yPos || 0) + 4 +  rowHeight/2)
+        .attr("x", (d) => 11)
+        .attr("y", (d) =>  (d.data.ontology &&  d.data.ontology.faUnicode.length === 3 ? 3 : 4) +  rowHeight/2)
```
Comment (dptelicent) [inline src/components/OntologyHierarchy/OntologyHierarchy.tsx:260]:
```suggestion
        .attr("y", (d) =>  (d.data.ontology?.faUnicode.length === 3 ? 3 : 4) +  rowHeight/2)
```

File: src/components/OntologyHierarchy/OntologyHierarchy.tsx [hunk 3]
```diff
@@ -186,36 +244,32 @@ const OntologyHierarchy: React.FC<OntologyHierarchyProps> = ({
...
         .style("fill", (d) => !d.data.ontology ? "" : d.data.ontology.color)
-        .attr("font-size", 10)
+        .attr("font-size", (d) => d.data.ontology &&  d.data.ontology.faUnicode.length === 3 ? 8 : 10)
```
Comment (dptelicent) [inline src/components/OntologyHierarchy/OntologyHierarchy.tsx:262]:
```suggestion
        .attr("font-size", (d) => d.data.ontology?.faUnicode.length === 3 ? 8 : 10)
```

File: src/components/OntologyHierarchy/OntologyHierarchyDisplay.tsx [hunk 4]
```diff
@@ -8,11 +8,27 @@ const OntologyHierarchyDisplay = (inputs: { instanceId: string }) => {
...
         data={hierarchy}
+        descendantCount={false}
+        dragEvent={handleDrag}
+        isDraggable={true}
+        filterIds={[
+            "http://ies.data.gov.uk/ontology/ies4#RealEstate",
+            "http://ies.data.gov.uk/ontology/ies4#PersonState",
+            "http://ies.data.gov.uk/ontology/ies4#ParticularPeriod",
+        ]}
+        expandAll={true}
```
Comment (dptelicent) [inline src/components/OntologyHierarchy/OntologyHierarchyDisplay.tsx:30]:
```suggestion
        isDraggable
        filterIds={[
            "http://ies.data.gov.uk/ontology/ies4#RealEstate",
            "http://ies.data.gov.uk/ontology/ies4#PersonState",
            "http://ies.data.gov.uk/ontology/ies4#ParticularPeriod",
        ]}
        expandAll
```

**Review by dptelicent — DISMISSED** [review]

**Review by ACoolmanTelicent — APPROVED** [review]

**Review by dptelicent — APPROVED** [review]

**Pull request merged**